### PR TITLE
feat(config): add config for Zooz ZEN51 and ZEN52 Smart Relays

### LINKED ISSUE #4119: Missing device configuration: Zooz ZEN52
### Checklist

- [X] It is not in the [configuration DB](https://devices.zwave-js.io/?search=3)

- [X] It was not [merged recently](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Apr+is%3Aclosed) or has a [pending PR](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Aopen+is%3Apr)

### Which device is missing?

Zooz ZEN52

### Manufacturer ID

0x027a

### Product Type

0x0104

### Product ID

0x0202

### Firmware Version

1.11

### Is the device listed on the Z-Wave Alliance website?

_No response_

### Do you have a manual?

[https://cdn.shopify.com/s/files/1/0218/7704/files/zooz-700-series-z-wave-plus-double-relay-zen52-manual.pdf](https://cdn.shopify.com/s/files/1/0218/7704/files/zooz-700-series-z-wave-plus-double-relay-zen52-manual.pdf)

### LINKED ISSUE #4393: Missing device configuration: <zooz zen51>
### Checklist

- [X] It is not in the [configuration DB](https://devices.zwave-js.io/?search=3)

- [X] It was not [merged recently](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Apr+is%3Aclosed) or has a [pending PR](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Aopen+is%3Apr)

### Which device is missing?

Zooz zen51

### Manufacturer ID

Zooz

### Product Type

0x0104

### Product ID

0x0201

### Firmware Version

1.10

### Is the device listed on the Z-Wave Alliance website?

_No response_

### Do you have a manual?

https://www.getzooz.com/downloads/zooz-700-series-z-wave-plus-dry-contact-relay-zen51-manual.pdf

### PR DESCRIPTION
**Pull request by zetlen**
Closes #4119.
Fixes: https://github.com/zwave-js/node-zwave-js/issues/4393

- Adds common parameter data for Zooz's new relays to Zooz manufacturer template.
- Adds config file for [ZEN52 Double Relay](https://www.thesmartesthouse.com/products/zooz-700-series-z-wave-plus-double-relay-zen52)
- Adds config file for [ZEN51 Dry Contact Relay](https://www.thesmartesthouse.com/products/zooz-700-series-z-wave-plus-dry-contact-relay-zen51) as well, since it has so many common parameters with the ZEN52.

### notes
- Manuals are linked directly in `metadata` section.
- As of this PR, these devices are not in the Z-Wave Alliance product database yet, so there is no entry in the config files for `zwaveAllianceId`.

### CODE REVIEW

**Comment by zetlen** [comment]
Both config files are tested on the real hardware.

**Comment by AlCalzone** [comment]
As a heads up, I've asked @blhoward2 to review this but he's swamped at work. Might take a few days.

File: packages/config/config/devices/0x027a/templates/zooz_template.json [hunk 1]
```diff
@@ -252,6 +276,88 @@
...
+			{
+				"label": "Momentary switch",
+				"value": 1
+			},
+			{
+				"label": "Toggle switch with fixed actions",
+				"value": 2
+			},
+			{
+				"label": "3-way impulse control",
```
Comment (blhoward2) [inline packages/config/config/devices/0x027a/templates/zooz_template.json]: What does 3-way impulse control mean?

File: packages/config/config/devices/0x027a/templates/zooz_template.json [hunk 2]
```diff
@@ -252,6 +276,88 @@
...
+				"value": 3
+			},
+			{
+				"label": "Garage door mode (simulate momentary)",
```
Comment (blhoward2) [inline packages/config/config/devices/0x027a/templates/zooz_template.json]: Just say "Garage door more (momentary)"

File: packages/config/config/devices/0x027a/templates/zooz_template.json [hunk 3]
```diff
@@ -252,6 +276,88 @@
...
+	},
+	"impulse_duration": {
+		"label": "Impulse Duration",
+		"description": "Choose the duration of the impulse if you're using this relay with 3-way impulse control",
+		"valueSize": 1,
+		"minValue": 2,
+		"maxValue": 200,
+		"defaultValue": 10,
+		"unsigned": true,
+		"unit": "seconds"
```
Comment (blhoward2) [inline packages/config/config/devices/0x027a/templates/zooz_template.json]: Minor nit, but please put the unit after valueSize.

File: packages/config/config/devices/0x027a/templates/zooz_template.json [hunk 4]
```diff
@@ -252,6 +276,88 @@
...
+		"label": "Relay Type",
+		"valueSize": 1,
+		"minValue": 0,
+		"maxValue": 1,
+		"defaultValue": 0,
+		"unsigned": true,
+		"allowManualEntry": false,
+		"options": [
+			{
+				"label": "Normally open (reports off when relay is open / on when relay is closed)",
```
Comment (blhoward2) [inline packages/config/config/devices/0x027a/templates/zooz_template.json]: Please change to "Normally open (off when relay is open)" The other way can be inferred.

File: packages/config/config/devices/0x027a/zen51.json [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+		},
+		{
+			"#": "10",
+			"$import": "templates/zooz_template.json#auto_timer_unit_select",
+			"label": "Timer Unit"
+		},
+		{
+			"#": "4",
+			"$import": "~/templates/master_template.json#state_after_power_failure_off_on_prev",
+			"label": "State After Power Failure"
```
Comment (blhoward2) [inline packages/config/config/devices/0x027a/zen51.json]: There is already a label in the master template. No need to override it (and please don't where a label already exists).

File: packages/config/config/devices/0x027a/zen51.json [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+		},
+		{
+			"#": "3",
+			"$import": "templates/zooz_template.json#auto_timer_variable_unit_base",
+			"label": "Auto On Timer"
+		},
+		{
+			"#": "10",
+			"$import": "templates/zooz_template.json#auto_timer_unit_select",
+			"label": "Timer Unit"
```
Comment (blhoward2) [inline packages/config/config/devices/0x027a/zen51.json]: There is already a label in the template you're adding, right? Please just use make the template the one and use that one.

File: packages/config/config/devices/0x027a/zen51.json [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+			"$import": "~/templates/master_template.json#state_after_power_failure_off_on_prev",
+			"label": "State After Power Failure"
+		},
+		{
+			"#": "5",
+			"$import": "templates/zooz_template.json#enable_scene_control"
+		},
+		{
+			"#": "6",
+			"$import": "~/templates/master_template.json#smart_switch_mode_0-2",
```
Comment (blhoward2) [inline packages/config/config/devices/0x027a/zen51.json]: Remove any other unnecessary labels, throughout, please.

**Review by blhoward2 — DISMISSED** [review]
Good work so far. Some questions and requested changes.

File: packages/config/config/devices/0x027a/templates/zooz_template.json (hunk 1 above)
Comment (zetlen) [reply packages/config/config/devices/0x027a/templates/zooz_template.json]: These are pretty new devices and the documentation from Zooz is still a little sparse. According to the [advanced settings article here](https://www.support.getzooz.com/kb/article/796-zen51-dry-contact-relay-advanced-settings/):

>(for selected 3-way scenarios, refer to the notes in the wiring instructions for your set-up)

However, the manuals don't explain this; instead they have QR codes that link back to the knowledge base article above. A circular reference!

I reckon, based on the features of other relays like the PAN03 or the Aeotec Nanos, that this setting makes the relay look for a short current pulse from a circuit like the ones depicted here: https://help.aeotec.com/support/solutions/articles/6000176583-nano-dimmer-wiring-3-way-4-way-5-way-switch-methods-

These are for scenarios where the line is in the wrong gang box for a standard 3-way setup. Unfortunately I haven't tested these cases, so I can't be sure that Zooz means the same thing that Aeotec means with their relays. I could open a ticket with them, I suppose, but I don't think I'd get around to opening up my wall and testing it.

In the meantime, should I just add Zooz's frustraing blurb about referring to other wiring instructions that don't exist yet? Or should I paraphrase what I said above, in as few words as possible?

**Comment by zetlen** [comment]
Not sure what the policy on this project is for turnaround or "bump" comments. I just want to know if there's anything I forgot to do to re-request review!

**Comment by AlCalzone** [comment]
Thanks for the reminder - blhoward is quite busy recently, so maybe I'll review the latest set of changes when I find the time.

File: packages/config/config/devices/0x027a/templates/zooz_template.json (hunk 1 above)
Comment (AlCalzone) [reply packages/config/config/devices/0x027a/templates/zooz_template.json]: I'd just play the ball back to Zooz and refer to the device manual / wiring instructions. We should probably play nice and tell them that they are nowhere to be found.

**Comment by AlCalzone** [comment]
All changes look good, except for the impulse control stuff.

File: packages/config/config/devices/0x027a/zen51.json [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+			"$import": "templates/zooz_template.json#relay_type_behavior"
+		},
+		{
+			"#": "11",
+			"$import": "templates/zooz_template.json#impulse_duration"
+		}
+	],
+	"metadata": {
+		"$import": "templates/zooz_template.json#zen5x_metadata",
+		"manual": "https://cdn.shopify.com/s/files/1/0218/7704/files/zooz-700-series-z-wave-plus-double-relay-zen52-manual.pdf"
```
Comment (towerhand) [inline packages/config/config/devices/0x027a/zen51.json]: This is the link for the ZEN51 manual
https://cdn.shopify.com/s/files/1/0218/7704/files/zooz-700-series-z-wave-plus-dry-contact-relay-zen51-manual.pdf

**Comment by AlCalzone** [comment]
@zetlen not sure if you got the notification

**Comment by zetlen** [comment]
I did, thanks. I haven't opened the Zooz support ticket yet. When I do, I'll update this PR.

File: packages/config/config/devices/0x027a/zen51.json (hunk 8 above)
Comment (zetlen) [reply packages/config/config/devices/0x027a/zen51.json]: Whoops, thanks. I'll correct it.

**Comment by AlCalzone** [comment]
Ok just ping me when you're done.

**Comment by zetlen** [comment]
Received word back from Zooz support, @AlCalzone. They plan to publish updated docs in two weeks, including up-to-date wiring diagrams; I thought it best to wait until then, if that's all right.

At minimum, they want to change "3-way impulse control" to just "impulse control". Excerpted:

>This is our first product with a more complex functionality around 3-way so we're in the process of rethinking how to phrase this setting accurately. It would be great if the name of the setting in the JS library didn't include the word "3-way" at all but instead said "impulse control". It applies to 3-way set-up where load and line are in separate boxes only, as shown in the attached diagram. We hope to have the diagrams up in the knowledge base in the next couple of weeks so the QR codes will finally work and lead to a full explanation. In many 3-way scenarios where load and line connections are in the same box, no parameter changes are needed but in this particular type, impulse control needs to be enabled.
>
>The reason we want to avoid the word "3-way" in the setting name is to prevent users from enabling impulse control for the 3-way scenarios where it's not needed.

(It strikes me that this could be described as "reverse momentary", since it detects a brief interruption in the NO state of the switch input. 🤔 )

**Comment by blhoward2** [comment]
Impulse control sounds terribly opaque. I'm fine with it not saying 3-way, but we generally try to make the labels descriptive so people understand what they actually do without resorting to the manual.

**Comment by zetlen** [comment]
I think just "Pulse" would be simpler and clearer, but I can see why it's tough to describe. The transient pulse between one contact and the other is a little bit harder for an amateur electrician to think about. That's why I think this should just link directly to the diagram.

**Comment by blhoward2** [comment]
I'll defer to @AlCalzone here. Personally, the function of this parameter is for 3way devices and so I'd say that in the label even if Zooz doesn't like it. Maybe 'Split 3-Way" (since the line/load are in different boxes) to differentiate it from the regular 3-way devices. I don't personally like impulse control as no one will understand what that means.

File: packages/config/config/devices/0x027a/zen52.json [hunk 9]
```diff
@@ -0,0 +1,125 @@
+{
+	"manufacturer": "Zooz",
+	"manufacturerId": "0x027a",
+	"label": "ZEN52",
+	"description": "Z-Wave Plus 700 Series Double Relay",
+	"devices": [
+		{
+			"productType": "0x0104",
+			"productId": "0x0202"
```
Comment (oxo42) [inline packages/config/config/devices/0x027a/zen52.json]: FWIW [there is now a ZWave Alliance ID of 4609](https://products.z-wavealliance.org/products/4609)

**Review by oxo42 — CHANGES_REQUESTED** [review]
There is now a Z-Wave Alliance Product, [4609](https://products.z-wavealliance.org/products/4609)

File: packages/config/config/devices/0x027a/zen51.json [hunk 10]
```diff
@@ -0,0 +1,77 @@
+{
+	"manufacturer": "Zooz",
+	"manufacturerId": "0x027a",
+	"label": "ZEN51",
+	"description": "Z-Wave Plus 700 Series Dry Contact Relay",
+	"devices": [
+		{
+			"productType": "0x0104",
+			"productId": "0x0201"
```
Comment (blakek) [inline packages/config/config/devices/0x027a/zen51.json]:
```suggestion
			"productId": "0x0201",
			"zwaveAllianceId": 4610
```

File: packages/config/config/devices/0x027a/zen52.json (hunk 9 above)
Comment (blakek) [reply packages/config/config/devices/0x027a/zen52.json]:
```suggestion
			"productId": "0x0202",
			"zwaveAllianceId": 4609
```

**Comment by AlCalzone** [comment]
@zetlen @blhoward2 
I like "Split 3-Way", maybe even mention in the description for that param that this is meant for setups where line and load are in different boxes.
Please also add the IDs that @blakek mentioned. Then we can merge this finally 💪🏻

**Comment by blhoward2** [comment]
I sent Zooz a note telling them we would be calling these "Split 3-Way"

**Comment by zetlen** [comment]
Good news, everyone! Zooz got back to me; they've added wiring diagrams and detailed instructions for both of these relays.

https://www.support.getzooz.com/kb/article/857-3-way-wiring-diagrams-for-the-zen51-dry-contact-relay/

https://www.support.getzooz.com/kb/article/858-3-way-wiring-diagrams-for-the-zen52-double-relay/

@blhoward2 I've updated the text to say "Split 3-Way". The diagram confirms that load/line split is the exact use case. Let me know if there's a nice place to put links to these diagrams; I suspect not, since the QR code in the manual now links to the folder containing the diagrams above.

**Comment by AlCalzone** [comment]
@zetlen At the moment we don't have any fields for custom URLs. If the manual now links there, I'd probably refer to the manual instead.

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**